platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManager.java

```diff
@@ -125,7 +126,9 @@ public void setInitialState(@NonNull final MerkleRoot state) {
     public void handleConsensusRound(final ConsensusRound round) {
         final MerkleRoot state = stateRef.get();
 
-        uptimeTracker.handleRound(round, state.getReadablePlatformState().getAddressBook());
+        uptimeTracker.handleRound(
+                round,
+                RosterRetriever.buildRoster(state.getReadablePlatformState().getAddressBook()));
```

Comment:
The `handleConsensusRound` is something that executes rather often, while the roster (currently) doesn't change that often. Building a new instance for a (currently) immutable object every time a round is processed seems expensive from the performance perspective.

Let's find a way to avoid having this translation here.

Reply:
fair point.  let me look at the best way to avoid this.